contracts/sui/xcall/sources/centralized_connection/centralized_state.move

```diff
@@ -73,6 +73,8 @@ module xcall::centralized_state {
 
     public(package) fun check_save_receipt(self: &mut State, net_id: String, sn: u128) {
         let receipt_key = ReceiptKey { nid: net_id, conn_sn: sn };
+        let (oldest_key, value) = self.receipts.get_entry_by_idx(0);
+        assert!(net_id!=oldest_key.nid || sn>oldest_key.conn_sn,90);
```

Comment:
ya this is problematic because if current nid is not equal to oldest key nid then older packets can also be submitted? we would need to save max sn by nid then only this would be able to prevent replay of old packets.